src/java/picard/vcf/FixVcfHeader.java

```diff
@@ -0,0 +1,199 @@
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package picard.vcf;
+
+import htsjdk.samtools.util.CloserUtil;
+import htsjdk.samtools.util.IOUtil;
+import htsjdk.samtools.util.Log;
+import htsjdk.samtools.util.ProgressLogger;
+import htsjdk.variant.variantcontext.Genotype;
+import htsjdk.variant.variantcontext.VariantContext;
+import htsjdk.variant.variantcontext.writer.Options;
+import htsjdk.variant.variantcontext.writer.VariantContextWriter;
+import htsjdk.variant.variantcontext.writer.VariantContextWriterBuilder;
+import htsjdk.variant.vcf.VCFFileReader;
+import htsjdk.variant.vcf.VCFFormatHeaderLine;
+import htsjdk.variant.vcf.VCFHeader;
+import htsjdk.variant.vcf.VCFHeaderLine;
+import htsjdk.variant.vcf.VCFHeaderLineCount;
+import htsjdk.variant.vcf.VCFHeaderLineType;
+import htsjdk.variant.vcf.VCFInfoHeaderLine;
+import htsjdk.variant.vcf.VCFStandardHeaderLines;
+import picard.PicardException;
+import picard.cmdline.CommandLineProgram;
+import picard.cmdline.CommandLineProgramProperties;
+import picard.cmdline.Option;
+import picard.cmdline.StandardOptionDefinitions;
+import picard.cmdline.programgroups.VcfOrBcf;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Tool for replacing or fixing up a VCF header.
+ *
+ * @author Nils Homer
+ */
+@CommandLineProgramProperties(
+        usage = FixVcfHeader.USAGE_SUMMARY + FixVcfHeader.USAGE_DETAILS,
+        usageShort = FixVcfHeader.USAGE_SUMMARY,
+        programGroup = VcfOrBcf.class
+)
+public class FixVcfHeader extends CommandLineProgram {
+    static final String USAGE_SUMMARY = "Replaces or fixes a VCF header.";
+    static final String USAGE_DETAILS = "Replaces or fixes a VCF heade.<br />" +
+            "This tool will either replace the header in the input file with the given VCF header or will try to fixup the input header. " +
+            "In the latter case, two-passes over the input file are required, and any INFO and FORMAT fields found in the VCF " +
+            "records but not found in the VCF header will be added to the output VCF header.<br />" +
+            "<h4>Replace header usage example:</h4>" +
+            "<pre>" +
+            "java -jar picard.jar FixVcfHeader \\<br />" +
+            "     I=input.vcf \\<br />" +
+            "     O=fixed.vcf \\<br />" +
+            "     HEADER=header.vcf" +
```

Comment:
Can HEADER be a full vcf or should it contain only the header that we want to use as base for fixing?